### ISSUE TITLE
Error while installing diff-gaussian-rasterization

### ISSUE THREAD

**Issue by ThomasDa** [dreamgaussian/dreamgaussian#58]
Hi everyone,

pip install ./diff-gaussian-rasterization failed with an Error : file not found

`Processing c:\ai\dream\diff-gaussian-rasterization
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: diff-gaussian-rasterization
  Building wheel for diff-gaussian-rasterization (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      No CUDA runtime is found, using CUDA_HOME='C:\Users\Thomas\.conda\envs\dream'
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\diff_gaussian_rasterization
      copying diff_gaussian_rasterization\__init__.py -> build\lib.win-amd64-cpython-311\diff_gaussian_rasterization
      running build_ext
      C:\ProgramData\anaconda3\Lib\site-packages\torch\utils\cpp_extension.py:383: UserWarning: Error checking compiler version for cl: [WinError 2] Le fichier spécifié est introuvable
        warnings.warn(f'Error checking compiler version for {compiler}: {error}')
      error: [WinError 2] Le fichier spécifié est introuvable
      [end of output]
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for diff-gaussian-rasterization
  Running setup.py clean for diff-gaussian-rasterization
Failed to build diff-gaussian-rasterization
ERROR: Could not build wheels for diff-gaussian-rasterization, which is required to install pyproject.toml-based projects
`

I work with a french Windows 10 OS, so "Le fichier spécifié est introuvable" means "The specified file can not be found".

I followed all the steps presented here: https://www.youtube.com/watch?v=_hy6iCq9f9o
The "git clone --recursive https://github.com/ashawkey/diff-gaussian-rasterization" worked correctly.
Do you have any idea what went wrong ?

Thanks in advance

**Comment by eointolster** [comment]
You need to have cuda installed https://developer.nvidia.com/cuda-toolkit
Then possibly 
conda install -y -c "nvidia/label/cuda-12.1.0" cuda-runtime

pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121

**Comment by andzejsp** [comment]
im having same problem on ubuntu, what am i missing?

**Comment by ThomasDa** [comment]
Thanks !

It seems that I installed the wrong version of CUDA (12.3 instead of 12.1) because I got "No CUDA runtime is found, using CUDA_HOME=....". 

@andzejsp your version is 11.5, I think you need to install 12.1, right?

I reinstall the correct version and try again (CUDA is 3GB my Internet connexion is slow ^^)

**Comment by andzejsp** [comment]
> Thanks !
> 
> It seems that I installed the wrong version of CUDA (12.3 instead of 12.1) because I got "No CUDA runtime is found, using CUDA_HOME=....".
> 
> @andzejsp your version is 11.5, I think you need to install 12.1, right?
> 
> I reinstall the correct version and try again (CUDA is 3GB my Internet connexion is slow ^^)

Is there a way to install that cuda on ubuntu and not breaking the current cuda? I remember breaking my cuda versioons doing some updates and in the end my stabledifussion stopped working :(

**Comment by andzejsp** [comment]
@ThomasDa thanks for the info

**Comment by andzejsp** [comment]
@ThomasDa on my windows 11 i had problem no cuda runtime found, using CUDA_HOME... And it turns out i had to install pytorch + cuda
```
pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121
```
https://pytorch.org/get-started/locally/

**Comment by CGMikeG** [comment]
Hello all,

I tried installing below version of pytorch and it work for me.

conda install pytorch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 pytorch-cuda=11.8 -c pytorch -c nvidia
https://pytorch.org/get-started/previous-versions/

Make sure you use the correct version that matches your cuda version

**Comment by sirdystically** [comment]
Similar issue here but with finding Torch

D:\dreamgaussian-main>pip install ./diff-gaussian-rasterization
Processing d:\dreamgaussian-main\diff-gaussian-rasterization
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "C:\Program Files\Python310\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Program Files\Python310\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "C:\Program Files\Python310\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "C:\Users\tony\AppData\Local\Temp\pip-build-env-qaoa08nt\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "C:\Users\tony\AppData\Local\Temp\pip-build-env-qaoa08nt\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "C:\Users\tony\AppData\Local\Temp\pip-build-env-qaoa08nt\overlay\Lib\site-packages\setuptools\build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\tony\AppData\Local\Temp\pip-build-env-qaoa08nt\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 13, in <module>
      ModuleNotFoundError: No module named 'torch'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by sirdystically** [comment]
But from same command line:

D:\dreamgaussian-main>pip show torch
Name: torch
Version: 2.2.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: c:\program files\python310\lib\site-packages
Requires: filelock, fsspec, jinja2, networkx, sympy, typing-extensions
Required-by: accelerate, torchaudio, torchvision

**Comment by sirdystically** [comment]
(CUDA 12.3 is installed and CUDA_HOME set if this matters)

**Comment by fty2000** [comment]
> Reference in new issue

thx！！！

**Comment by DongChengdongHangZhou** [comment]
> Hello all,
> 
> I tried installing below version of pytorch and it work for me.
> 
> conda install pytorch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 pytorch-cuda=11.8 -c pytorch -c nvidia https://pytorch.org/get-started/previous-versions/
> 
> Make sure you use the correct version that matches your cuda version

Thank you, CGMikeG, your suggestion really helps me.

My ubuntu machine is 4090 and cuda 12.2, and I meet this problem when I install diff-gaussian-rasterization.
After I run your command: 
conda install pytorch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 pytorch-cuda=11.8 -c pytorch -c nvidia
I have successfully installed the diff-gaussian-rasterization.

**Comment by Dusk-Hermit** [comment]
You may need to check if your version of PyTorch is the GPU version. I didn't notice that I had installed the CPU version, which could lead to errors like the one below.

(dreamgaussian) D:\repos\dreamgaussian>pip install ./diff-gaussian-rasterization
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Processing d:\repos\dreamgaussian\diff-gaussian-rasterization
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "D:\repos\dreamgaussian\diff-gaussian-rasterization\setup.py", line 21, in <module>
          CUDAExtension(
        File "D:\software\anaconda3\envs\dreamgaussian\lib\site-packages\torch\utils\cpp_extension.py", line 1078, in CUDAExtension
          library_dirs += library_paths(cuda=True)
        File "D:\software\anaconda3\envs\dreamgaussian\lib\site-packages\torch\utils\cpp_extension.py", line 1212, in library_paths
          paths.append(_join_cuda_home(lib_dir))
        File "D:\software\anaconda3\envs\dreamgaussian\lib\site-packages\torch\utils\cpp_extension.py", line 2420, in _join_cuda_      OSError: CUDA_HOME environment variable is not set. Please set it to your CUDA install root.

error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
